File: apps/dashboard/frontend/src/models/plan.ts
import { PlanItem } from "./plan-item";

export interface Plan {
  id: string;
  name: string;
  currency: string;
  status: string;
  trial_period_days: number;
  created_at: string; // ISO date string
  updated_at: string; // ISO date string
  items: PlanItem[];
  interval: string;
  interval_count: number;
  plan_group: any;
  features: any[];
  metered_features: any[];
}

export interface UpdatePlanRequest {
  name: string;
  currency: string;
  trial_period_days: number;
  interval: string;
  interval_count: number;
  plan_group_id: string;
}

File: libs/go/apix/plan.go
package apix

import (
	"time"

	"github.com/useportcall/portcall/libs/go/dbx/models"
)

type Plan struct {
	ID              string     `json:"id"`
	Name            string     `json:"name"`
	Currency        string     `json:"currency"`
	Status          string     `json:"status"`
	TrialPeriodDays int        `json:"trial_period_days"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
	Items           []PlanItem `json:"items"`
	Interval        string     `json:"interval"`
	IntervalCount   int        `json:"interval_count"`
	PlanGroup       any        `json:"plan_group"`
	Features        []any      `json:"features"`
	MeteredFeatures []any      `json:"metered_features"`
}

func (p *Plan) Set(plan *models.Plan) *Plan {
	p.ID = plan.PublicID
	p.Name = plan.Name
	p.Currency = plan.Currency
	p.Status = plan.Status
	p.TrialPeriodDays = plan.TrialPeriodDays
	p.CreatedAt = plan.CreatedAt
	p.UpdatedAt = plan.UpdatedAt
	p.Interval = plan.Interval
	p.IntervalCount = plan.IntervalCount
	return p
}
